lib/common/containers/ActiveUserRetriever.js

```diff
@@ -0,0 +1,41 @@
+// @flow
+import { useAuth0 } from '@auth0/auth0-react'
+// $FlowFixMe useEffect not recognized by flow.
+import { useEffect } from 'react'
+import { connect } from 'react-redux'
+
+import * as userActions from '../../manager/actions/user'
+
+/**
+ * Retrieves the active, authenticated user from auth0 and
+ * updates the redux state accordingly.
+ */
+const ActiveUserRetriever = ({ logout, receiveTokenAndProfile }) => {
+  const auth0 = useAuth0()
+  const { isAuthenticated, isLoading, user: profile } = auth0
+
+  // Any time the login state reported by auth0 changes,
+  // update the user info in the redux state.
+  useEffect(async () => {
+    if (isAuthenticated) {
+      const tokenResponse = await auth0.getAccessTokenSilently({
+        detailedResponse: true
+      })
+      // TRICKY: What we are passing to the backend is the id_token portion of the response.
+      // Unlike the access_token portion, the id_token contains the app_metadata info for the user.
+      receiveTokenAndProfile({ profile, token: tokenResponse.id_token })
+    } else if (!isLoading) {
```

Comment:
I think `isLoading` might have some unstable behavior -- I'm seeing the `USER_LOGGED_OUT` redux action firing right after I log in (on Firefox). The result is that the "Learn more about data tools" disappears, but the "Existing users [sign in here](http://localhost:9966/login)." does not disappear!

Reply:
No errors in the console I should add

Reply:
The logout action that happens here is equivalent of the one [here](https://github.com/ibi-group/datatools-ui/blob/cc70292c3f9172253e7dbfd477a381d43cad8419/lib/common/user/Auth0Manager.js#L303) and is meant to handle cases when `isAuthenticated` is false (e.g. a login expires) and auth0 is not doing a check in the background for the login status (`isLoading` is false)